src/sentry/integrations/github/tasks/link_all_repos.py

```diff
@@ -35,52 +40,64 @@ def link_all_repos(
     integration_id: int,
     organization_id: int,
 ):
-    integration = integration_service.get_integration(
-        integration_id=integration_id, status=ObjectStatus.ACTIVE
-    )
-    if not integration:
-        logger.error(
-            "%s.link_all_repos.integration_missing",
-            integration_key,
-            extra={"organization_id": organization_id},
-        )
-        metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
-        return
 
-    rpc_org = organization_service.get(id=organization_id)
-    if rpc_org is None:
-        logger.error(
-            "%s.link_all_repos.organization_missing",
-            integration_key,
-            extra={"organization_id": organization_id},
-        )
-        metrics.incr(
-            f"{integration_key}.link_all_repos.error",
-            tags={"type": "missing_organization"},
+    with SCMIntegrationInteractionEvent(
+        interaction_type=SCMIntegrationInteractionType.LINK_ALL_REPOS,
+        provider_key=integration_key,
+        organization=organization_id,
+    ).capture() as lifecycle:
+        integration = integration_service.get_integration(
+            integration_id=integration_id, status=ObjectStatus.ACTIVE
         )
-        return
-
-    installation = integration.get_installation(organization_id=organization_id)
-
-    client = installation.get_client()
+        if not integration:
+            # TODO: Remove this logger in favor of context manager
+            logger.error(
+                "%s.link_all_repos.integration_missing",
+                integration_key,
```

Comment:
I've been wondering about this: What's the difference between `integration_key` and `integration.provider`?

Reply:
ah this is a very good question. there *should* be the same in this context are two different purposes here. `integration_key` is used in the install pipeline where this task is queued. 

the integration.provider is generally used to grab the integration installation which we can use to hit to make api calls. 

we should probably consolidate them or i think the reason they aren't is because we won't have a provider until after the install pipeline is completed?

Reply:
Oh that mostly makes sense, but wouldn't it only apply to installations? I think we track those with an entirely separate metric: https://github.com/getsentry/sentry/blob/a213dc8ea45b0ba00f7e449fe42adab559452252/src/sentry/integrations/utils/metrics.py#L271